view/sharedcache/core/SharedCache.cpp

```diff
@@ -2779,6 +2779,31 @@ std::vector<Ref<Symbol>> SharedCache::ParseExportTrie(std::shared_ptr<MMappedFil
 	return symbols;
 }
 
+
+std::vector<std::pair<uint64_t, std::pair<BNSymbolType, std::string>>> SharedCache::GetExportListForHeader(SharedCacheMachOHeader header, std::function<std::shared_ptr<MMappedFileAccessor>()> provideLinkeditFile)
+{
+	if (auto it = m_exportInfos.find(header.textBase); it != m_exportInfos.end())
+	{
+		return it->second;
+	}
+	else
+	{
+		std::shared_ptr<MMappedFileAccessor> linkeditFile = provideLinkeditFile();
+		if (!linkeditFile)
+			return std::vector<std::pair<uint64_t, std::pair<BNSymbolType, std::string>>>();
+
+		auto exportList = SharedCache::ParseExportTrie(linkeditFile, header);
+		std::vector<std::pair<uint64_t, std::pair<BNSymbolType, std::string>>> exportMapping(exportList.size());
```

Comment:
Since this is now the only caller of `ParseExportTrie`, it'd be beneficial to have it produce data in the format that this expects. This will eliminate the need for `ParseExportTrie` / `ReadExportNode` to allocate a bunch of `Symbol` instances only for this code to copy data out of them and then discard them. All those memory allocations and copying all of the names around is surprisingly expensive.

Reply:
Yes that was lazy of me, I'll add another commit to do that